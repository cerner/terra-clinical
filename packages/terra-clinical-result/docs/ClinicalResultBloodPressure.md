# Terra Clinical Result Blood Pressure

The Clinical Result is a standardized view for single result values added to a Patient's Medical Chart. This is specifically for Blood Pressure Vital Sign results.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
```
$ npm install terra-clinical-result
```

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
<!-- AUTO-GENERATED-CONTENT:END -->

## Component Features

* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
* [LTR/RTL Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#ltr--rtl-support)

## Usage Guide

The blood pressure result is constructed similarly to the single clinical result by providing structured object with a specific construction that follows a similar pattern to the [HL7 FHIR Observation Blood Pressure](https://www.hl7.org/fhir/observation-example-bloodpressure.html) example.

An example of a single blood pressure result value:
```jsx
import React from 'react';
import { ClinicalResultBloodPressure } from 'terra-clinical-result/lib/index';

const bloodpressureResultValue = {
  id: '1577836800',
  systolic: {
    eventId: '1577836800.1',
    result: {
      value: '180',
      unit: 'mmHg',
    },
    interpretation: 'critical',
    isModified: true,
    hasComment: true,
    isUnverified: true,
    conceptDisplay: 'Blood Pressure',
    datetimeDisplay: 'Nov 23, 2019 13:31:31',
  },
  diastolic: {
    eventId: '1577836800.2',
    result: {
      value: '60',
      unit: 'mmHg',
    },
    interpretation: 'low',
    isModified: true,
    hasComment: true,
    isUnverified: true,
    conceptDisplay: 'Blood Pressure',
    datetimeDisplay: 'Nov 23, 2019 13:31:31',
  }
};

export default () => <ClinicalResultBloodPressure {...bloodpressureResultValue} hideUnit isTruncated />;

```
