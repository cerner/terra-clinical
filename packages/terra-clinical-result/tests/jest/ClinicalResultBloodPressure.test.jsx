import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import ClinicalResultBloodPressure from '../../src/ClinicalResultBloodPressure';
import { DefaultBloodPressureResult, DefaultSystolicResult, DefaultDiastolicResult } from '../../src/terra-dev-site/test/clinical-result/TestResults';

describe('ClinicalResultBloodPressure', () => {
  it('should render a ResultError if hasResultError is true', () => {
    const result = shallowWithIntl(<ClinicalResultBloodPressure hasResultError />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a NoData if hasResultNoData is true', () => {
    const result = shallowWithIntl(<ClinicalResultBloodPressure hasResultNoData />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a default ClinicalResultBloodPressure', () => {
    const result = shallowWithIntl(<ClinicalResultBloodPressure {...DefaultBloodPressureResult} />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultBloodPressure', () => {
    const result = shallowWithIntl(<ClinicalResultBloodPressure {...DefaultBloodPressureResult} isTruncated />).dive();
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultBloodPressure with icons moved', () => {
    const resultData = {
      id: '111',
      systolic: {
        ...DefaultSystolicResult,
        isModified: true,
      },
      diastolic: {
        ...DefaultDiastolicResult,
        isModified: true,
      },
    };
    const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} isTruncated />).dive();
    expect(result).toMatchSnapshot();
  });

  describe('- Systolic -', () => {
    it('should pass certain props to Observation ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          interpretation: 'critical',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} hideUnit />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a modified result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          isModified: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a commented result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          hasComment: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render an unverified result ', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          isUnverified: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with unit hidden', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} hideUnit />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a concept display', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          conceptDisplay: 'Temperature Oral',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a date time display', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          datetimeDisplay: 'Nov 23, 2019 13:31:31',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a entered-in-error status result', () => {
      const resultData = {
        id: '111',
        systolic: {
          ...DefaultSystolicResult,
          status: 'entered-in-error',
          interpretation: 'critical',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });
  });

  describe('- Diastolic -', () => {
    it('should pass certain props to Observation ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          interpretation: 'critical',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} hideUnit />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a modified result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          isModified: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a commented result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          hasComment: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render an unverified result ', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          isUnverified: true,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with unit hidden', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} hideUnit />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a concept display', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          conceptDisplay: 'Temperature Oral',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a result with a date time display', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          datetimeDisplay: 'Nov 23, 2019 13:31:31',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });

    it('should render a entered-in-error status result', () => {
      const resultData = {
        id: '111',
        diastolic: {
          ...DefaultDiastolicResult,
          status: 'entered-in-error',
          interpretation: 'critical',
        },
      };
      const result = shallowWithIntl(<ClinicalResultBloodPressure {...resultData} />).dive();
      expect(result).toMatchSnapshot();
    });
  });
});
