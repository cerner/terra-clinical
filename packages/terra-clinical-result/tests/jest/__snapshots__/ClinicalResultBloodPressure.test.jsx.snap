// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicalResultBloodPressure - Diastolic - should pass certain props to Observation  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={true}
          interpretation="critical"
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a commented result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
      <IconComment
        className="icon-comment"
        data-name="Layer 1"
        isBidi={true}
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a entered-in-error status result 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display status-in-error"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        condition={true}
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          interpretation={null}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a modified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
      <IconModified
        className="icon-modified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with a concept display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
  <div
    className="concept-display"
  >
    Temperature Oral
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with a date time display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
  <div
    className="concept-display"
  >
    Nov 23, 2019 13:31:31
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render a result with unit hidden 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={true}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Diastolic - should render an unverified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <InjectIntl(ResultError)
        key="Error-Systolic-111"
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.2"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          isUnverified={true}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
      <IconDiamond
        className="icon-unverified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should pass certain props to Observation  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={true}
          interpretation="critical"
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a commented result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
      <IconComment
        className="icon-comment"
        data-name="Layer 1"
        isBidi={true}
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a entered-in-error status result 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display status-in-error"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={true}
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          interpretation={null}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a modified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
      <IconModified
        className="icon-modified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with a concept display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Temperature Oral
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with a date time display 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
    </div>
  </div>
  <div
    className="concept-display"
  >
    Nov 23, 2019 13:31:31
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render a result with unit hidden 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={true}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure - Systolic - should render an unverified result  1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        key="del-Systolic-111.1"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.1"
          hideUnit={false}
          isUnverified={true}
          key="Observation-Systolic-111.1"
          result={
            Object {
              "unit": "mmHg",
              "value": "180",
            }
          }
        />
      </ConditionalWrapper>
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <InjectIntl(ResultError)
        key="Error-Diastolic-111"
      />
      <IconDiamond
        className="icon-unverified"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a NoData if hasResultNoData is true 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <InjectIntl(KnownNoData) />
</div>
`;

exports[`ClinicalResultBloodPressure should render a ResultError if hasResultError is true 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <InjectIntl(ResultError) />
</div>
`;

exports[`ClinicalResultBloodPressure should render a default ClinicalResultBloodPressure 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        key="Observation-Systolic-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a truncated ClinicalResultBloodPressure 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        key="Observation-Systolic-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
  </div>
</div>
`;

exports[`ClinicalResultBloodPressure should render a truncated ClinicalResultBloodPressure with icons moved 1`] = `
<div
  className="clinical-result blood-pressure-result"
>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <Observation
        eventId="111.1"
        hideUnit={true}
        key="Observation-Systolic-111.1"
        result={
          Object {
            "unit": "mmHg",
            "value": "180",
          }
        }
      />
      <span
        className="result-display-separator"
        key="Observation-Separator-111.1"
      >
        /
      </span>
      <ConditionalWrapper
        key="del-Diastolic-111.2"
        wrapper={[Function]}
      >
        <Observation
          eventId="111.2"
          hideUnit={false}
          key="Observation-Diastolic-111.2"
          result={
            Object {
              "unit": "mmHg",
              "value": "80",
            }
          }
        />
      </ConditionalWrapper>
    </div>
    <IconModified
      className="icon-modified"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
</div>
`;
